src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProvidersConfig.java

```diff
@@ -121,6 +121,11 @@ interface ModelConfig {
 
       @JsonProperty
       Map<String, String> properties();
+
+      /** Flag to mark if support for the model is deprecated. */
+      @Nullable
+      @JsonProperty
+      Optional<Boolean> deprecated();
```

Comment:
Do we need to use `Optional` here? I know it's "optional" in protoc but should be fine to leave this as regular `Boolean`, as it defaults to `false` (which I think is proper default for our usage).

Reply:
Makes sense. Changed it to regular `Boolean` in this commit https://github.com/stargate/data-api/pull/1625/commits/62a48fc3348211dc56f545f749d2e9b38d9f2707

Reply:
Reverted this change, since it expects the existing configs to have this property set up
```
java.util.NoSuchElementException: SRCFG00014: The config property stargate.jsonapi.embedding.providers.cohere.models[0].deprecated is required but it could not be found in any config source
```

Reply:
Ah. I wonder what component fails (Quarkus build?) -- but if it is needed then it is needed.